### PR TITLE
Remove block parameter from querying methods

### PR DESCRIPTION
**Pull request by riley-klingler**
Some of these methods (e.g. `left_joins`) do not take a block at all. Others (e.g. `select`) can take a block, but `sorbet` is not expressive enough to [properly typecheck](https://github.com/chanzuckerberg/sorbet-rails/issues/152) calls where a block is passed in.

I checked all of these functions and think it's correct that either of those two conditions are true for all of them, but checking was fairly repetitive so it would be great if someone could check my work.

### CODE REVIEW

**Comment by manhhung741** [comment]
Thank you Riley! I'll review the PR asap!

File: lib/sorbet-rails/model_plugins/active_record_querying.rb [hunk 1]
```diff
@@ -33,7 +33,6 @@ def generate(root)
         method_name.to_s,
         parameters: [
           Parameter.new("*args", type: "T.untyped"),
-          Parameter.new("&block", type: "T.nilable(T.proc.void)"),
```
Comment (manhhung741) [inline lib/sorbet-rails/model_plugins/active_record_querying.rb]: I think all querying methods are defined here:

https://github.com/rails/rails/blob/9a7f15c905c9ca7a26b56b6f2a045cf56805a99e/activerecord/lib/active_record/relation/query_methods.rb

I did a grep of block and find that other than `select`, `extending` is the only other method accepting a block
https://github.com/rails/rails/blob/9a7f15c905c9ca7a26b56b6f2a045cf56805a99e/activerecord/lib/active_record/relation/query_methods.rb#L918

Can you separate it from the rest?

Comment (riley-klingler) [reply lib/sorbet-rails/model_plugins/active_record_querying.rb]: Thanks for catching that!

Comment (manhhung741) [reply lib/sorbet-rails/model_plugins/active_record_querying.rb]: Looks good now! thanks for fixing!

**Pull request merged**